lib/remote-objects.js

```diff
@@ -707,22 +708,90 @@ RemoteObjects.prototype.getScope = function(ctx, method) {
  * Define a named type conversion. The conversion is used when a
  * `SharedMethod` argument defines a type with the given `name`.
  *
- * ```js
- * remotes.defineType('MyType', function(val, ctx) {
- *   // use the val and ctx objects to return the concrete value
- *   return new MyType(val);
+ * ```
+ * remotes.defineType('MyType', {
+ *   fromTypedValue: function(ctx, value) {
+ *     // use the val and ctx objects to return the concrete value
+ *     if (value === undefined || value === null)
+ *       return { value: value };
+ *
+ *     value = new MyType(value);
+ *     var error = this.validate(ctx, value);
+ *     return error ? { error: error } : { value: value };
+ *   },
+ *
+ *   fromSloppyValue: function(ctx, value) {
+ *     // use the val and ctx objects to return the concrete value
+ *     // apply any coercion needed to convert values coming from
+ *     // string-only sources like HTTP headers and query string.
+ *     var objectConverter = ctx.typeRegistry.getConverter('object');
+ *     var result = objectConverter.fromSloppyString(ctx, value);
+ *     return result.error ? result : this.fromTypedValue(ctx, result.value);
+ *   },
+ *
+ *   validate: function(ctx, value) {
+ *     if (value === undefined || value === null)
+ *       return null;
+ *     if (typeof value !== 'object' || !value instanceof MyType) {
+ *       return new Error('Value is not an instance of MyType.');
+ *     }
+ *     return null;
+ *   },
+ * });
+ *
+ * See also `remotes.defineObjectType`.
+ *
+ * @param {String} name The type name
+ * @param {Object} converter A converter implementing the following methods:
+ *   - `fromTypedValue(ctx, value) -> { value } or { error }`
+ *   - `fromSloppyValue(ctx, value) -> { value } or { error }`
+ *   - `validate(ctx, value) -> error or undefined/null`
+ */
+
+RemoteObjects.prototype.defineType = function(name, converter) {
+  if (typeof converter === 'function') {
+    throw new Error(g.f(
+      '%s is no longer supported. Use one of the new APIs instead: %s or %s',
+      'remoteObjects.defineType(name, fn)',
+      'remoteObjects.defineObjectType(name, factoryFn)',
+      'remoteObjects.defineType(name, converter)'));
+  }
+  this._typeRegistry.defineType(name, converter);
+};
+
+/**
+ * Define a named type conversion for an object-like type.
+ * The conversion is used when a `SharedMethod` argument
+ * defines a type with the given `name`.
+ *
+ * ```
+ * remotes.defineObjectType('MyClass', function(data) {
+ *   return new MyClass(data);
  * });
  * ```
  *
- * **Note: the alias `remotes.convert()` is deprecated.**
+ * Under the hood, a converter is created that ensures the input data
+ * is an object (or sloppy value is coerced to an object) and calls
```

Comment:
See https://github.com/strongloop/strong-remoting/pull/343#discussion_r77302252

> The sloppy conversion/coercion is invoked when the value comes from a string-only source like HTTP header or more importantly from a query string parsed via qs.

Should I expand this comment to make it more clear what is meant by sloppy values?